test/syscalls/linux/pty.cc

```diff
@@ -1384,6 +1384,25 @@ TEST_F(PtyTest, PartialBadBuffer) {
   EXPECT_THAT(munmap(addr, 2 * kPageSize), SyscallSucceeds()) << addr;
 }
 
+// Test that writing nothing to the PTY replica's output queue does not return an error.
+TEST_F(PtyTest, ReplicaWriteNothingCanonical) {
+  constexpr char kInput[] = "";
+  EXPECT_THAT(WriteFd(replica_.get(), kInput, strlen(kInput)),
+              SyscallSucceedsWithValue(strlen(kInput)));
```

Comment:
should this be `strlen` or `sizeof`?

Reply:
`strlen` seems right. `sizeof` would write a null byte '\0'.

Reply:
Sounds good. My other PR used `sizeof`: https://github.com/google/gvisor/pull/9954/files#r1475132056 which is why I'm unsure.